module/gui/CaptchaDialog.py

```diff
@@ -0,0 +1,257 @@
+
+        self.imgLabel  = QLabel()
+        self.infoLabel = QLabel()
+        self.infoLabel.setAlignment(Qt.AlignCenter)
+        self.lineEdit  = QLineEdit()
+
+        self.buttons = WtDialogButtonBox(Qt.Horizontal, self)
+        self.buttons.hideWhatsThisButton()
+        self.submitBtn = self.buttons.addButton(QDialogButtonBox.Ok)
+        self.ignoreBtn = self.buttons.addButton(QDialogButtonBox.Ignore)
+        self.closeBtn  = self.buttons.addButton(QDialogButtonBox.Close)
+        self.buttons.button(QDialogButtonBox.Ok).setText(_("Submit"))
+        self.buttons.button(QDialogButtonBox.Ignore).setText(_("Ignore"))
+        self.buttons.button(QDialogButtonBox.Close).setText(_("Close"))
+        self.ignoreBtn.setFocusPolicy(Qt.NoFocus)   # do not close the dialog on return keypress
+        self.closeBtn.setFocusPolicy(Qt.NoFocus)    # do not close the dialog on return keypress
+
+        hbox = QHBoxLayout()
+        hbox.addStretch()
+        hbox.addWidget(self.imgLabel)
+        hbox.addStretch()
+        vbox = QVBoxLayout()
+        vbox.addLayout(hbox)
+        vbox.addWidget(self.infoLabel)
+        vbox.addWidget(self.lineEdit)
+        vbox.addSpacing(7)
+        vbox.addStretch()
+        vbox.addLayout(self.buttons.layout())
+        self.setLayout(vbox)
+
+        self.connect(self.submitBtn, SIGNAL("clicked()"),       self.slotSubmitText)
+        self.connect(self.lineEdit,  SIGNAL("returnPressed()"), self.slotSubmitText)
+        self.connect(self.ignoreBtn, SIGNAL("clicked()"),       self.slotIgnore)
+        self.connect(self.closeBtn,  SIGNAL("clicked()"),       self.slotClose)
+        self.imgLabel.mousePressEvent = self.slotSubmitPos
+
+        self.connect(self, SIGNAL("setTask"), self.setTask)
+        self.connect(self, SIGNAL("setMovie(QMovie *)"), self.imgLabel, SLOT("setMovie(QMovie *)"))
+        self.connect(self, SIGNAL("show"), self.slotShow)
+        self.connect(self, SIGNAL("slotShow_continue"), self.slotShow_continue, Qt.QueuedConnection)
+        self.connect(self, SIGNAL("setFree"), self.setFree)
+
+        self.setMinimumWidth(250)
+
+    def isFree(self):
+        return not self.processing
+
+    def setTask(self, tid, data, type, resultType):
+        self.processing = True
+        self.currentID = tid
+        self.currentResultType = resultType
+
+        self.lineEdit.setText("")
+        self.infoLabel.setText("")
+        errMsg = None
+
+        if self.imgAnimation != None and self.imgAnimation.isValid():
+            self.imgAnimation.stop()
+        if self.imgDataBuffer != None and self.imgDataBuffer.isOpen():
+            self.imgDataBuffer.close()
+
+        (dummy, encoded) = data['src'].split(",", 1)   # separate the metadata from the image data
+        self.imgData = QByteArray.fromBase64(encoded)
+        self.imgDataBuffer = QBuffer(self.imgData)
+        if self.imgDataBuffer.open(QIODevice.ReadOnly):
+            self.imgAnimation = QMovie()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['QMovie' may be undefined, or defined from star imports: PyQt4.QtCore, PyQt4.QtGui](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=567903)